src/vm.rs

```diff
@@ -439,6 +439,9 @@ pub(crate) fn run(
                     if option_flags & OPTION_TRACE != 0 {
                         println!("saves: {:?}", state.saves);
                     }
+                    if state.saves[0] > state.saves[1] {
+                        state.save(0, state.saves[1]);
+                    }
```

Comment:
not sure if it's okay to access `saves` by index here, or if we should use `get` instead?
```suggestion
                    let slot1 = state.get(1);
                    if state.get(0) > slot1 {
                        state.save(0, slot1);
                    }
```

Reply:
Yeah let's use `get` like in other places.